### PR TITLE
Refactor widgets, restyle RSS widget

### PR DESCRIPTION
**Pull request by noisecapella**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots 
  - [x] tag @ferdi or @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Part of #1640 

#### What's this PR do?
Refactors widgets so that a react component is required for every widget type and that JSON is provided via the backend instead of prerendered HTML. 

#### How should this be manually tested?
Add an RSS widget. It should render properly and look like the mockup

#### Screenshot
screenshot from 2019-01-17 10-13-41

### CODE REVIEW

File: widgets/serializers/rss.py [hunk 1]
```diff
@@ -31,33 +31,35 @@ class RssFeedWidgetSerializer(WidgetInstanceSerializer):
...
-                    '<p><a href="{entry_link}">{entry_timestamp} | {entry_title}<a><p>',
-                    entry_link=entry_link,
-                    entry_timestamp=entry_timestamp,
-                    entry_title=entry_title,
-                )
-            )
-        return "\n".join(formatted_items)
+
+        return {
+            "title": rss.feed.title,
```
Comment (mbertrand) [inline widgets/serializers/rss.py]: I think the RSS widget title entered by the user should be used if not blank?

Comment (noisecapella) [reply widgets/serializers/rss.py]: :+1:

**Comment by noisecapella** [comment]
Just realized I missed writing tests for `RssWidget` and `UrlWidget`, I'll add them before merging

**Comment by mbertrand** [comment]
ok, though it looks like `lib/widgets_test.js` is hitting 100% coverage of them.

**Pull request merged**